spec/classes/github_actions_runner_spec.rb

```diff
@@ -49,32 +40,100 @@
+            'ensure' => 'present',
+            'user'   => 'root',
+            'group'  => 'root',
+          })
+        end
+      end
+
+      context 'is expected to contain an ownership exec' do
+        it do
+          is_expected.to contain_exec('first_runner-ownership').with({
+            'user'    => 'root',
+            'command' => '/bin/chown -R root:root /some_dir/actions-runner-2.272.0/first_runner',
+          })
+        end
+      end
+
+      context 'is expected to contain a Run exec' do
+        it do
+          is_expected.to contain_exec('first_runner-run_configure_install_runner.sh').with({
+            'user'    => 'root',
+            'command' => '/some_dir/actions-runner-2.272.0/first_runner/configure_install_runner.sh',
+          })
+        end
+      end
+
       context 'is expected to create a github_actions_runner installation script' do
         it do
-          is_expected.to contain_file('/tmp/actions-runner-1.0.1/first_runner/configure_install_runner.sh').with({
+          is_expected.to contain_file('/some_dir/actions-runner-2.272.0/first_runner/configure_install_runner.sh').with({
             'ensure' => 'present',
-            'owner'  => 'test_user',
-            'group'  => 'test_group',
+            'owner'  => 'root',
+            'group'  => 'root',
             'mode'   => '0755',
           })
         end
       end
 
       context 'is expected to create a github_actions_runner installation script with config in content' do
         it do
-          is_expected.to contain_file('/tmp/actions-runner-1.0.1/first_runner/configure_install_runner.sh').with_content(/\/tmp\/actions-runner-1.0.1\/first_runner\/config.sh/)
+          is_expected.to contain_file('/some_dir/actions-runner-2.272.0/first_runner/configure_install_runner.sh').with_content(/\/some_dir\/actions-runner-2.272.0\/first_runner\/config.sh/)
         end
       end
 
       context 'is expected to create a github_actions_runner installation script with repo url in content' do
         it do
-          is_expected.to contain_file('/tmp/actions-runner-1.0.1/first_runner/configure_install_runner.sh').with_content(/https:\/\/github.com\/test_org\/test_repo/)
+          is_expected.to contain_file('/some_dir/actions-runner-2.272.0/first_runner/configure_install_runner.sh').with_content(/https:\/\/github.com\/github_org\/test_repo/)
         end
       end
 
       context 'is expected to create a github_actions_runner installation script with labels in content' do
         it do
-          is_expected.to contain_file('/tmp/actions-runner-1.0.1/first_runner/configure_install_runner.sh').with_content(/test_label1,test_label2/)
+          is_expected.to contain_file('/some_dir/actions-runner-2.272.0/first_runner/configure_install_runner.sh').with_content(/test_label1,test_label2/)
+        end
+      end
+    end
+  end
+end
+
+
+describe 'github_actions_runner' do
```

Comment:
Is this the test name ? If so, I would use another name for this other test.

Reply:
no its the class name have to be the same . The problem is im not sure its the best way to add another example (test case) in the same file.

But i didnt manage to do so with another context or another file.